scripts/data/gen-nodetime/src/relayer/lib/relayer.ts

```diff
@@ -1,141 +1,215 @@
+        const dstClient = await Relayer.getIBCClient(dstChain, dstKey);
+        const link = await this.create(srcClient, dstClient, clientIdA, clientIdB);
+
+        const channels = await link.createChannel(
+            'A',
+            path.src.port_id,
+            path.dst.port_id,
+            orderFromJSON(path.ordering),
+            path.dst.version
+        );
+
+        path.src.channel_id = channels.src.channelId;
+        path.dst.channel_id = channels.dest.channelId;
+        path.src.connection_id = link.endA.connectionID;
+        path.dst.connection_id = link.endB.connectionID;
+
+        return path;
+    }
+
+    public async start([
+                           path,
+                           srcChain,
+                           dstChain,
+                           srcKey,
+                           dstKey
+                       ]: [Path, Chain, Chain, string, string]): Promise<Path> {
+        const srcClient = await Relayer.getIBCClient(srcChain, srcKey);
+        const dstClient = await Relayer.getIBCClient(dstChain, dstKey);
+
+        const link = await Link.createWithExistingConnections(
+            srcClient,
+            dstClient,
+            path.src.connection_id,
+            path.dst.connection_id,
+            new ConsoleLogger()
+        );
+
+        const heights = await link.checkAndRelayPacketsAndAcks(
+            {
+                packetHeightA: path.src.packet_height,
+                packetHeightB: path.dst.packet_height,
+                ackHeightA: path.src.ack_height,
+                ackHeightB: path.dst.ack_height
+            } ?? {},
+            2,
+            6
+        );
+
+        await link.updateClientIfStale('A', this.defaultMaxAge);
+        await link.updateClientIfStale('B', this.defaultMaxAge);
+
+        path.src.packet_height = heights.packetHeightA;
+        path.dst.packet_height = heights.packetHeightB;
+        path.src.ack_height = heights.ackHeightA;
+        path.dst.ack_height = heights.ackHeightB;
+
+        return path;
+    }
+
+    private static async getIBCClient(chain: Chain, key: string): Promise<IbcClient> {
+        const chainGP = GasPrice.fromString(chain.gas_price);
+        const signer = await DirectSecp256k1Wallet.fromKey(fromHex(key), chain.address_prefix);
+
+        const [account] = await signer.getAccounts();
+
+        return await IbcClient.connectWithSigner(
+            chain.rpc_address,
+            signer,
+            account.address,
+            {
+                prefix: chain.address_prefix,
+                gasPrice: chainGP,
+                gasLimits: calcGasLimits(chain.gas_limit)
+            }
+        );
+    }
+
+    public async create(
+        nodeA: IbcClient,
+        nodeB: IbcClient,
+        clientA?: string,
+        clientB?: string
+    ): Promise<Link> {
+
+        let clientIdB = clientB;
+        if (typeof clientIdB === 'undefined' || clientIdB === '') {
```

Comment:
```suggestion
        if (!srcClientID) {
```

does this work? - trying to simplify